### PR TITLE
Topologies monitoring. I21087

### PR DESCRIPTION
**Pull request by demonodojo**
Topologies monitoring. Extend API #21087

### CODE REVIEW

**Comment by amatosc** [comment]
Please resolve the conflicts before.

**Review by fandrod — APPROVED** [review]
La parte correspondiente al front end la veo correcta

File: sinfonier-backend-api/actions/topologies/GetTopologyInfo.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+class GetTopologyInfo(ApiBase):
+    @falcon.before(ApiBase.validate_params)
+    def on_get(self, req, resp, id):
+        topology_id = self.validated_params['id']
+        try:
+            topology = MongodbFactory.get_topology(topology_id)
+            if topology is None:
+                return ResponsesHandler.handle_404(resp, 'Topology not found')
+
+            info = StormUI.getTopologyByName(topology["name"])
```
Comment (amatosc) [inline sinfonier-backend-api/actions/topologies/GetTopologyInfo.py]: Please, use the constants defined in SinfonierConstants.py

File: sinfonier-backend-api/actions/topologies/GetTopologyInfo.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+            return ResponsesHandler.handle_200(resp, {'id': topology_id, 'info': info})
+        except HTTPError as ex:
+            logger.error(ex.message)
+            if ex.response.status_code == 400:
+                return ResponsesHandler.handle_200(resp, {'id': topology_id, 'log': ''})
```
Comment (amatosc) [inline sinfonier-backend-api/actions/topologies/GetTopologyInfo.py]: Why do you return an exception as 200 code?

File: sinfonier-web/public/javascripts/topologies.js [hunk 3]
```diff
@@ -488,4 +488,91 @@ $(function(){
...
+		        		              {"title":"Capacity (last 10m)","data":"capacity"},
+		        		              {"title":"Execute latency (ms)","data":"executeLatency"},
+		        		              {"title":"Executed","data":"executed"},
+		        		              {"title":"Process latency (ms)","data":"processLatency"},
+		        		              {"title":"Acked","data":"acked"},
+		        		              {"title":"Failed","data":"failed"},
+		        		              {"title":"Error Host","data":"errorHost"},
+		        		              {"title":"Error Port","data": "errorPort"},
+		        		              {"title":"Last Error","data": "lastError"}]
+	    	  				,res.data.bolts);
```
Comment (amatosc) [inline sinfonier-web/public/javascripts/topologies.js]: Please, use i18n for make it available for the current languages.

File: sinfonier-web/public/javascripts/topologies.js [hunk 4]
```diff
@@ -488,4 +488,91 @@ $(function(){
...
+      });
+  };
+  //Info
+  $('.collapse-info').click(function(){
+	  var target = $(this).attr('href');
+	  var $target = $(target);
+	  var $info_div = $target.find('.info-expanded');
+	  $info_div.html(''); 
+	  if (!$target.is(':visible'))
+	  {
```
Comment (amatosc) [inline sinfonier-web/public/javascripts/topologies.js]: please, use a correct format.

**Review by amatosc — CHANGES_REQUESTED** [review]
I can't see the test in sinfonier-backend-api and sinfonier-web neither. As well check other comments in the code please.

File: sinfonier-backend-api/actions/topologies/GetTopologyInfo.py (hunk 2 above)
Comment (demonodojo) [reply sinfonier-backend-api/actions/topologies/GetTopologyInfo.py]: This is an error

File: sinfonier-web/test/tests/controllers/TopologiesTest.java [hunk 5]
```diff
@@ -27,6 +37,30 @@ public static void beforeClass() throws Exception {
...
+    }
+  }
+  
+  public static void unmockClient() {
+    try {
+      Field field = Topologies.class.getDeclaredField("client");
+      field.setAccessible(true);
+      field.set(null, Client.getInstance());
+    } catch (Exception e) {
+      e.printStackTrace();
```
Comment (amatosc) [inline sinfonier-web/test/tests/controllers/TopologiesTest.java:63]: if this exception is caught what will happen? I think that have to raise it.

File: sinfonier-web/test/tests/controllers/TopologiesTest.java [hunk 6]
```diff
@@ -27,6 +37,30 @@ public static void beforeClass() throws Exception {
...
+
+  public static void mockClient() {
+    Client client =  TestMock.mockTopologiesClient();
+    if (client != null) {
+      try {
+        Field field = Topologies.class.getDeclaredField("client");
+        field.setAccessible(true);
+        field.set(null, client);
+      } catch (Exception e) {
+        e.printStackTrace();
```
Comment (amatosc) [inline sinfonier-web/test/tests/controllers/TopologiesTest.java:51]: if this exception is caught what will happen? I think that have to raise it.

**Review by amatosc — COMMENTED** [review]
Please check the comments.

**Review by amatosc — APPROVED** [review]

**Review by ajsanchezsanz — APPROVED** [review]

**Pull request merged**